master/pkg/stream/stream.go

```diff
@@ -81,24 +97,38 @@ func (s *Subscription[T]) Configure(filter func(T) bool) {
 			s.Publisher.Subscriptions = s.Publisher.Subscriptions[:last]
 			break
 		}
-	} else {
-		// Modify an existing registraiton.
-		// (just save filter, below)
-	}
+	} // else modify an existing registration, update subscription filter
+
 	// Remember the new filter.
 	s.filter = filter
 }
 
+// Publisher is responsible for publishing messages of type T
+// to streamers associate with active subscriptions.
 type Publisher[T Msg] struct {
-	Lock sync.Mutex
+	Lock          sync.Mutex
 	Subscriptions []*Subscription[T]
-	WakeupID int64
+	WakeupID      int64
 }
 
-func NewPublisher[T Msg]() *Publisher[T]{
+// NewPublisher creates a new Publisher for message type T.
+func NewPublisher[T Msg]() *Publisher[T] {
 	return &Publisher[T]{}
 }
 
+// Restart closes all streamers associated with this Publisher.
+func (p *Publisher[T]) Restart() {
```

Comment:
can we rename this `p.CloseAllStreamers` or `p.CloseAllSubscribers` or something?

From the name I would think the name `Restart()` suggests that this would have similar behavior to `NewPublisher()`, but that's not the case.

Reply:
This method hasn't been renamed, even though you have a :+1: and marked the comment as resolved.

Reply:
I renamed `p.Restart() -> p.CloseAllStreamers()`; are you also talking about the PublisherSet's `Restart()` as well?